### PR TITLE
Week 2

### PR DESCRIPTION
**Pull request by carlota-moh**
Currently working on hooks and routing. There is still a funny bug preventing index.tsx from loading

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -8,5 +8,5 @@
 <body>
     <div id="root"></div>
 </body>
-<script src="/src/index.tsx"></script>
+<script type="module" src="./src/index.tsx"></script>
```
Comment (aemartinez37) [inline index.html:11]: Hey @carlota-moh 
I've been checking your reported issue and I have a question:
Why did you add this line to load the `./src/index.tsx`?

Try just removing this line and let's see if your issue is fixed, let me know what happens, thanks.

Comment (carlota-moh) [reply index.html:11]: Hi @aemartinez37 
Thanks for the input!! I suspected it had something to do with that, although I couldn't quite figure out why. I researched into it and I realized that the webpack HTML plugin was the one in charge of loading the compiled index into the HTML. Since browsers can't read TS, it was being interpreted as a JSON rather than an executable and it obviously could not locate it... Dumb mistake on my part, thanks for helping :)

File: jest.config.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import type {Config} from '@jest/types';
+// Sync object
+const config: Config.InitialOptions = {
+  verbose: true,
+  transform: {
+  '^.+\\.tsx?$': 'ts-jest',
+  },
+};
+export default config;
```
Comment (aemartinez37) [inline jest.config.ts:9]:
```suggestion
import type { Config } from "@jest/types";
// Sync object
const modulesToTransform = [
  "pokedex-promise-v2",
  "p-map",
  "aggregate-error",
  "indent-string",
  "clean-stack",
  "escape-string-regexp",
].join("|");
const config: Config.InitialOptions = {
  verbose: true,
  transform: {
    "^.+\\.(j|t)sx?$": "ts-jest",
  },
  transformIgnorePatterns: [`node_modules/(?!${modulesToTransform}/.*)`],
};
export default config;
```

This configuration let's jest include those packages that apparently were not included properly by default.

In addition you should set as `true` the prop `allowJs` in the `tsconfig.json` file.

**Review by aemartinez37 — COMMENTED** [review]
Hey @carlota-moh 

Please try the suggested fix to the jest issue, let me know how it goes, thanks.

File: jest.config.ts (hunk 2 above)
Comment (carlota-moh) [reply jest.config.ts:9]: It worked!! Thank you so much 🙇🏼‍♂️
